### ISSUE TITLE
Use vertical velocities of ocean model for vertical advection

### ISSUE THREAD

**Issue by johannesro** [OpenDrift/opendrift#22]

**Comment by johannesro** [comment]
vertical advection will become a feature of all 3d models, i.e. opendrift3d, oildrift3d, pelagicegg.

@knutfrode : I also thought to introduce a new model class, OceanDrift3D, similar to the class OceanDrift. The class is mostly meant to track water particles in 3D, allowing for vertical advection, turbulent mixing, and additional wind drag near the surface, but no particle-specific behaviour or buoyancy as for i.e. oil droplet or pelagic eggs.

**Comment by knutfrode** [comment]
Yes, this is also a good idea.

I have been thinking a little about naming convention for 2D regarding 3D,
which should be consistent. We have the same issue e.g. for oil drift,
where we now have "OpenOil" (which is 2D) and "OpenOil3D". Alternatively
names could be more explicit (OpenOil2D, OpenOil3D), or we could use the
convention that drift is in 3D is default unless 2D is specifically
mentioned, i.e. (OpenOil2D, OpenOil) and (OceanDrift2D, OceanDrift).
Even another alternative is to merge the 2D and 3D models, with possibility
to switch vertical advection and turbulence on/off with a config setting
(e.g. o.config['3d'] = True)

Summing up the 4 alternatives:
1) OceanDrift, OceanDrift3D
2) OceanDrift2D, OceanDrift3D
3) OceanDrift2D, OceanDrift
4) OceanDrift

I think I have a small preference for option 2, which give least
possibilities for misunderstandings.

2016-08-18 9:23 GMT+02:00 johannesro <notifications@github.com>:

**Comment by johannesro** [comment]
Option 4 could make the code quite complicated and a bit more unreadable, therefore I would also vote for option 2, which is more consistent than 1 and 3.

**Comment by msdelalama** [comment]
I also think option 2 is the clearest :)
